cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/SparkCqlEvaluatorArgs.java

```diff
@@ -91,4 +91,10 @@
     
     @Parameter(names = { "--debug" }, description = "Enables CQL debug logging")
     public boolean debug = false;
+
+    @Parameter(names = { "--disable-result-grouping" }, description = "Disable use of CQL parameters to group context results into separate rows", required = false)
+    public boolean disableResultGrouping = false;
+    
+    @Parameter(names = { "--key-parameter" }, description = "One or more parameter names that should be included in the parameters column for output rows that are generated.", required = false)
```

Comment:
Should this be called `--key-parameters`?

Reply:
I considered it, but the value of each parameter is a single string. You would specify multiple args to get multiple values. For example, `--key-parameter EndDate --key-parameter MinimumAge` reads better to me than `--key-parameters EndDate --key-parameters MinimumAge`. JCommander doesn't support list parsing out of the box for something like `--key-parameters EndDate,MinimumAge`. We could implement a custom arg parser for that ourselves if we wanted, but I've been just sticking to the core functionality.

Reply:
I got curious about this today since wanting a simple comma-separated list of values comes up a lot. It turns out that JCommander _does_ support this natively, but only for parameters that are List types. It doesn't work for Set parameters even if you implement a custom listConverter class. I changed the Set parameters to List parameters and updated the code to support the new type. I also updated the long names of some of the existing parameters to reflect their multi-valued nature and for consistency across parameters.